.NET/Microsoft.Recognizers.Text.DateTime/Utilities/AgoLaterUtil.cs

```diff
@@ -225,106 +229,94 @@ public static class AgoLaterUtil
+                    dayStr = match.Groups["day"].Value;
                 }
-
-                resultDateTime = DurationParsingUtil.ShiftDateTime(timex, referenceTime.AddDays(swift), true);
-
-                ((DateTimeResolutionResult)durationParseResult.Value).Mod = Constants.AFTER_MOD;
             }
-            else if (utilityConfiguration.CheckBothBeforeAfter && (MatchingUtil.ContainsAgoLaterIndexInBeforeString(beforeStr, utilityConfiguration.LaterRegex) ||
-                     MatchingUtil.ContainsAgoLaterIndex(afterStr, utilityConfiguration.InConnectorRegex) ||
-                     MatchingUtil.ContainsAgoLaterIndex(afterStr, utilityConfiguration.LaterRegex)))
-            {
-                // Check also beforeStr
-                var match = utilityConfiguration.LaterRegex.Match(beforeStr);
-                var swift = 0;
 
-                if (MatchingUtil.ContainsAgoLaterIndex(afterStr, utilityConfiguration.LaterRegex) && string.IsNullOrEmpty(match.Groups["day"].Value))
-                {
-                    match = utilityConfiguration.LaterRegex.Match(beforeStr);
-                }
-
-                // Handle cases like "3 days after tomorrow"
-                if (match.Success && !string.IsNullOrEmpty(match.Groups["day"].Value))
+            if (isMatch)
+            {
+                // Handle cases like "3 days before yesterday", "3 days after tomorrow"
+                if (!string.IsNullOrEmpty(dayStr))
                 {
-                    swift = swiftDay(match.Groups["day"].Value);
+                    swift = swiftDay(dayStr);
                 }
 
-                var yearMatch = utilityConfiguration.SinceYearSuffixRegex.Match(beforeStr);
-                if (yearMatch.Success)
+                if (isLater)
                 {
-                    var yearString = yearMatch.Groups[Constants.YearGroupName].Value;
-                    var yearEr = new ExtractResult { Text = yearString };
-                    var year = Convert.ToInt32((double)(numberParser.Parse(yearEr).Value ?? 0));
-                    referenceTime = DateObject.MinValue.SafeCreateFromValue(year, 1, 1);
+                    var yearMatch = utilityConfiguration.SinceYearSuffixRegex.Match(afterStr);
+                    if (yearMatch.Success)
+                    {
+                        var yearString = yearMatch.Groups[Constants.YearGroupName].Value;
+                        var yearEr = new ExtractResult { Text = yearString };
+                        var year = Convert.ToInt32((double)(numberParser.Parse(yearEr).Value ?? 0));
+                        referenceTime = DateObject.MinValue.SafeCreateFromValue(year, 1, 1);
+                    }
                 }
 
-                resultDateTime = DurationParsingUtil.ShiftDateTime(timex, referenceTime.AddDays(swift), true);
+                resultDateTime = DurationParsingUtil.ShiftDateTime(timex, referenceTime.AddDays(swift), isLater ? true : false);

```

Comment:
Nit: I'd prefer assigning `isLater ? true : false` to a var and passing it as named argument in the method call.

Reply:
done